src/sys/mman.rs

```diff
@@ -1,241 +1,251 @@
-    pub const MADV_UNMERGEABLE: MmapAdvise  = 13; /* KSM may not merge identical pages.  */
-    pub const MADV_HUGEPAGE   : MmapAdvise  = 14; /* Worth backing with hugepages.  */
-    pub const MADV_NOHUGEPAGE : MmapAdvise  = 15; /* Not worth backing with hugepages.  */
-    pub const MADV_DONTDUMP   : MmapAdvise  = 16; /* Explicity exclude from the core dump, overrides the coredump filter bits.  */
-    pub const MADV_DODUMP     : MmapAdvise  = 17; /* Clear the MADV_DONTDUMP flag.  */
-    pub const MADV_HWPOISON   : MmapAdvise  = 100; /* Poison a page for testing.  */
-
-
-    libc_bitflags!{
-        pub flags MsFlags: c_int {
-            MS_ASYNC,
-            MS_INVALIDATE,
-            MS_SYNC,
-        }
-    }
-
-    pub const MAP_FAILED: isize               = -1;
-}
-
-#[cfg(any(target_os = "macos",
-          target_os = "ios"))]
-mod consts {
-    use libc::{self, c_int};
-
-    libc_bitflags!{
-        pub flags MapFlags: c_int {
-            MAP_FILE,
-            MAP_SHARED,
-            MAP_PRIVATE,
-            MAP_FIXED,
-            MAP_ANON,
-            MAP_NOCACHE,
-            MAP_JIT,
-        }
-    }
-
-    pub type MmapAdvise = c_int;
-
-    pub const MADV_NORMAL     : MmapAdvise      = 0; /* No further special treatment.  */
-    pub const MADV_RANDOM     : MmapAdvise      = 1; /* Expect random page references.  */
-    pub const MADV_SEQUENTIAL : MmapAdvise      = 2; /* Expect sequential page references.  */
-    pub const MADV_WILLNEED   : MmapAdvise      = 3; /* Will need these pages.  */
-    pub const MADV_DONTNEED   : MmapAdvise      = 4; /* Don't need these pages.  */
-    pub const MADV_FREE       : MmapAdvise      = 5; /* pages unneeded, discard contents */
-    pub const MADV_ZERO_WIRED_PAGES: MmapAdvise = 6; /* zero the wired pages that have not been unwired before the entry is deleted */
-    pub const MADV_FREE_REUSABLE : MmapAdvise   = 7; /* pages can be reused (by anyone) */
-    pub const MADV_FREE_REUSE : MmapAdvise      = 8; /* caller wants to reuse those pages */
-    pub const MADV_CAN_REUSE : MmapAdvise       = 9;
-
-    libc_bitflags!{
-        pub flags MsFlags: c_int {
-            MS_ASYNC, /* [MF|SIO] return immediately */
-            MS_INVALIDATE, /* [MF|SIO] invalidate all cached data */
-            MS_KILLPAGES, /* invalidate pages, leave mapped */
-            MS_DEACTIVATE, /* deactivate pages, leave mapped */
-            MS_SYNC, /* [MF|SIO] msync synchronously */
-        }
+libc_bitflags!{
+    /// Additional parameters for `mmap()`.
+    pub flags MapFlags: c_int {
+        /// Compatibility flag. Ignored.
+        MAP_FILE,
+        /// Share this mapping. Mutually exclusive with `MAP_PRIVATE`.
+        MAP_SHARED,
+        /// Create a private copy-on-write mapping. Mutually exclusive with `MAP_SHARED`.
+        MAP_PRIVATE,
+        /// Place the mapping at exactly the address specified in `addr`.
+        MAP_FIXED,
+        /// Synonym for `MAP_ANONYMOUS`. Deprecated.
```

Comment:
Probably only deprecated on Linux platforms. In fact it appears that on FreeBSD `MAP_ANON` is the "correct" constant and `MAP_ANONYMOUS` is just there for compatibility. I'll just remove the deprecation notice.